### PR TITLE
[Unit Test Improvement & Fixes] Card Details & Collection

### PR DESCRIPTION
**Pull request by mpetrenco**
Fixed, formatted and re-enabled Card Details and Collection unit tests;

### CODE REVIEW

File: Tests/APITests/CardDetailsSetTests.swift [hunk 1]
```diff
@@ -49,11 +67,21 @@ class CardDetailsSetTests : JudoTestCase {
...
+        XCTAssertEqual("**** **** **** 3436", cardInputField?.textField.text,
+                       "Card Input field does not match the masked value '**** **** **** 3436'")
+        
+        XCTAssertEqual("12/20", expiryDateInputField?.textField.text,
+                       "Expiry Date field does not match the value '12/20'")
     }
     
-    func testCardDetailsMustBeNotPresent() {
+    /**
+     * GIVEN: I have initialized the JudoPayViewController with valid no card details and
```
Comment (gcojocaru) [inline Tests/APITests/CardDetailsSetTests.swift]: What do you mean here ?

File: Tests/APITests/CardDetailsSetTests.swift [hunk 2]
```diff
@@ -64,6 +92,9 @@ class CardDetailsSetTests : JudoTestCase {
         XCTAssertEqual("", cardInputField?.textField.text)
```
Comment (gcojocaru) [inline Tests/APITests/CardDetailsSetTests.swift]: Maybe add failure description to keep it consistent.

Comment (mpetrenco) [reply Tests/APITests/CardDetailsSetTests.swift]: Done.

File: Tests/APITests/CardDetailsSetTests.swift [hunk 3]
```diff
@@ -73,8 +104,16 @@ class CardDetailsSetTests : JudoTestCase {
...
+        let vc = JudoPayViewController(judoId: myJudoId,
+                                       amount: oneGBPAmount,
+                                       reference: validReference,
+                                       transaction: .payment,
+                                       currentSession: judo,
+                                       cardDetails: cardDetails) { (_, _) in }
+        
         vc.paymentToken = paymentToken
         vc.theme = JPTheme()
         _ = vc.view
```
[the remark this answers is not in the document]
Comment (mpetrenco) [reply Tests/APITests/CardDetailsSetTests.swift]: Card details will not be set otherwise.

File: Tests/APITests/CardDetailsSetTests.swift (hunk 1 above)
Comment (mpetrenco) [reply Tests/APITests/CardDetailsSetTests.swift]: Fixed comment.

File: Tests/APITests/CollectionTests.swift [hunk 4]
```diff
@@ -27,54 +27,97 @@ import XCTest
...
     
-    func testCollection() {
+    /**
+     * GIVEN: I have made a pre-authorization request and received a receipt ID
+     *        and an amount
+     *
+     * WHEN:  I perform a collection with the receipt ID and the amount
+     *
+     * THEN:  I should receive a succesful response with at least one
+     *        JPTransactionData items
```
[the remark this answers is not in the document]
Comment (mpetrenco) [reply Tests/APITests/CollectionTests.swift]: Fixed comment.

**Pull request merged**